File: 백준/Silver/2635. 수 이어가기/수 이어가기.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n")

const n = Number(input[0])

const check = (num, num2) => {
  const tmp = [num, num2]
  let first = num
  let second = num2
  let third = first - second
  while (third >= 0) {
    tmp.push(third)
    first = second
    second = third
    third = first - second
  }
  return tmp
}
let answer = [n]
for (let i = n; i > 0; i -= 1) {
  const array = check(n,i)
  if(answer.length <= array.length){
    answer = array
  }
}
console.log(answer.length)
console.log(answer.join(' '))

File: 백준/Silver/2635. 수 이어가기/수 이어가기.py
import sys
input = sys.stdin.readline

n = int(input())

def cal(n1,n2):
  first = n1
  second = n2
  third = first-second
  result = [first,second]
  while(third >=0):
    result.append(third)
    first = second
    second = third
    third = first-second
  return result

answer =[]
for i in range(n,0,-1):
  array = cal(n,i)
  if(len(array) > len(answer)):
    answer = array

print(len(answer))
print(' '.join(list(map(str,answer))))